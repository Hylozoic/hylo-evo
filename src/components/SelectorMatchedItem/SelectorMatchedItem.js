import React from 'react'

import Icon from 'components/Icon'
import RoundImage from 'components/RoundImage'
import './SelectorMatchedItem.scss'

const { func, string } = React.PropTypes

export default function SelectorMatchedItem ({ avatarUrl, id, name, deleteMatch }) {
  return <div styleName='selector-matched-item'>
    <RoundImage url={avatarUrl} small styleName='avatar' />
    <span styleName='name'>{name}</span>
    <span onClick={deleteMatch}>
      <Icon name='Ex' styleName='delete-match' />
    </span>
  </div>
}

SelectorMatchedItem.propTypes = {
  name: string,
  deleteMatch: func
}
