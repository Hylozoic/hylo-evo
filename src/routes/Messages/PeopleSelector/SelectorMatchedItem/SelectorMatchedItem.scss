.selector-matched-item { 
  display: flex;
  align-items: center;
  margin-right: $space-2x;
  margin-top: $space-1x;
  padding: $space-1x;
  height: $space-10x;
  border: 1px $color-rhino-30 solid;
  border-radius: 4px;
}

.avatar {
  margin-right: $space-1x;
}

.name {
  composes: caption-drk-lg from 'css/typography.scss';
  @include font-bold;
}

.delete-match {
  font-size: 18px;
  color: $color-rhino-30;
  margin: 0 $space-2x;
  cursor: pointer;
}
