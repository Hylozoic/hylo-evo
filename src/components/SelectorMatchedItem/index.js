import SelectorMatchedItem from './SelectorMatchedItem'

export default SelectorMatchedItem
