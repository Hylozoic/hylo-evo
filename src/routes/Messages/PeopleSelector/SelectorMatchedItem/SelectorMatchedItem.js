import PropTypes from 'prop-types'
import React from 'react'

import Icon from 'components/Icon'
import RoundImage from 'components/RoundImage'
import './SelectorMatchedItem.scss'

const { func, string } = PropTypes

export default function SelectorMatchedItem ({ avatarUrl, id, name, removeParticipant }) {
  return <div styleName='selector-matched-item'>
    <RoundImage url={avatarUrl} small styleName='avatar' />
    <span styleName='name'>{name}</span>
    <span onClick={removeParticipant}>
      <Icon name='Ex' styleName='delete-match' />
    </span>
  </div>
}

SelectorMatchedItem.propTypes = {
  name: string,
  deleteMatch: func
}
