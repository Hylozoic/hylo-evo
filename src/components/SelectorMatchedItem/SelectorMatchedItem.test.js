import { mount, shallow } from 'enzyme'
import React from 'react'

import SelectorMatchedItem from './SelectorMatchedItem'

it('matches last snapshot', () => {
  const wrapper = shallow(<SelectorMatchedItem />)
  expect(wrapper).toMatchSnapshot()
})

it('calls deleteMatch when close button clicked', () => {
  const deleteMatch = jest.fn()
  const wrapper = mount(
    <SelectorMatchedItem deleteMatch={deleteMatch} />
  )
  wrapper.find('span').last().simulate('click')
  expect(deleteMatch).toHaveBeenCalled()
})
