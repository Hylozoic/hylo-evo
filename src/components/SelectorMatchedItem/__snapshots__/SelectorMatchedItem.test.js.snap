// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches last snapshot 1`] = `
<div
  data-styleName="selector-matched-item"
>
  <RoundImage
    data-styleName="avatar"
    small={true}
  />
  <span
    data-styleName="name"
  />
  <span>
    <Icon
      data-styleName="delete-match"
      name="Ex"
    />
  </span>
</div>
`;
